File: Programming Basics/JavaScript/Nested loops/Lab/combinations.js
function solve(input) {
    const n = Number(input[0]);

    // Mathematical solution using arithmetic progressions. Complexity: O(1)
    /*int ans = (n + 1) * (n + 2) / 2;
    console.log(ans);*/

    let ans = 0;

    for (let i = 0; i <= n; i++) {
        let remaining = n - i;
        for (let j = 0; j <= remaining; j++) {
            // i + j + _ = n
            ans++;
        }
    }

    console.log(ans);
}

File: Programming Basics/C++/Nested loops/Labs/Combinations/Combinations.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    cin >> n;

    // Mathematical solution using arithmetic progressions. Complexity: O(1)
    /*int ans = (n + 1) * (n + 2) / 2;
    cout << ans << endl;*/

    int ans = 0;

    for (int i = 0; i <= n; i++)
    {
        int remaining = n - i;
        for (int j = 0; j <= remaining; j++)
        {
            // i + j + _ = n
            ans++;
        }
    }

    cout << ans << endl;
}
